crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -380,6 +380,17 @@ def prepare_nodes_for_os_upgrade
       node.admin? || node[:platform] == "windows" || node.state != "crowbar_upgrade"
     end
     check_if_nodes_are_available upgrade_nodes
+
+    # reset the OS values at admin node...
+    admin_node = NodeObject.admin_node
+    admin_node["target_platform"] = ""
+    if admin_node["provisioner"] && admin_node["provisioner"]["default_os"]
+      admin_node["provisioner"].delete "default_os"
+    end
+    admin_node.save
+    # ... and let provisioner fill correct values for current OS
+    commit_and_check_proposal
+
```

Comment:
This (```prepare_nodes_for_os_upgrade```) really  seems to be the wrong place to do it. Doing it in the admin node upgrade script (like in the other PR) seems somewhat better, but I'd really like us to understand us to understand better why/when the attributes are repopulated and try to solve that, before papering over the issue in one or the other way.